soak/README.md

```diff
@@ -1,317 +1,117 @@
-
-
-### Step 7 (Cleanup Soak test chart)
-
-* RUN `helm -n ack-system uninstall $SOAK_RUNNER_CHART_RELEASE_NAME`
-
-### Step 8 (Cleanup the service controller deployment)
-
-* RUN `helm -n ack-system uninstall $CONTROLLER_CHART_RELEASE_NAME && kubectl delete namespace ack-system`
-
-### Step 9 (Optional: Cleanup the kube-prometheus-stack chart) 
-
-* RUN `helm -n prometheus uninstall $PROM_CHART_RELEASE_NAME && kubectl delete namespace prometheus`
-
-### Step 10 (Cleanup the background port-forward processes)
-
-* RUN 
-    ```bash
-    for pid in $(ps -a | grep "kubectl port-forward" | grep "$PROM_CHART_RELEASE_NAME" | cut -d" " -f1); do
-        kill -9 $pid ;
-    done
-    ```
\ No newline at end of file
+Create a new AWS account to house the following resources. A new account ensures
+that any leaked resources do not affect the e2e tests running on your pull
+requests, and that you can give the ACK core team permissions to view this
+account.
+
+Export the credentials for an AWS user with administrator access to your
+terminal before following the rest of the documents.
+
+### Step 1 (Run the bootstrapping script)
+
+Run the `bootstrap.sh` script in the current directory. This script will do the
+following: 
+1. Create an ECR public repository for the soak test runner
+1. Create an EKS cluster with the default soak test configuration (from
+  `cluster-config.yaml`)
+1. Install the controller Helm chart
+1. Install Prometheus and Grafana
+1. Install the custom ACK soak test Grafana dashboard (from
+   `./monitoring/grafana/ack-dashboard-source.json`)
+1. Build and push the soak test runner image
+1. Install the soak test runner Helm chart
+
+The script requires the name of the service and the semver tag of the controller
+version up for testing. For example:
+```bash
+./bootstrap.sh s3 v0.0.12
+```
+
+### Step 2 (Log onto the Prometheus dashboard)
+
+After the script concludes, it should provide an example of the command to
+port-forward Prometheus. By default, the command should look like the following:
+
+```bash
+kubectl port-forward -n prometheus service/kube-prom-grafana 3000:80 --address='0.0.0.0' >/dev/null &
+```
+
+Navigate to `http://127.0.0.1:3000/` and log in. You should be able to log into
+the dashboard with the following credentials:
+- Username: `admin`
+- Password: `prom-operator`
+
+Using the menu bar on the left, side navigate to `Dashboards` > `Browse`. Select
+the `ACK Dashboard` at the top of the list. The `ACK Dashboard` will show the
+ACK-specific request counts and error codes used by the controller when making
+calls to AWS APIs. 
+
+The `Kubernetes/Compute Resources/Pod` dashboard will show the resource
+consumption by the controller pod.
+
+### Step 3 (Notify the ACK core team)
+
+The ACK core team needs to be made aware that service controller is ready for
+soak test execution using Prow, which is a manual process. Therefore, once you
```

Comment:
TL;DR: I am in favor of supporting a separate soak-test account to run soak-tests from prow but it may not work straight away as mentioned in this doc.

-----------

Soak tests on prow currently have a dependency that EKS cluster should be created in the same account as e2e tests. See current onboarding instructions [here](https://github.com/aws-controllers-k8s/test-infra/tree/main/soak/prow) . It will need a small amount of work to decouple that. 

Currently, ACK post-submit-job assumes the e2e-test-iam-role(from ssm) before executing `eks update-kubeconfig` command for gaining access to customer's  EKS soak cluster. Also we ask customers to create their EKS soak cluster using `e2e-test-iam-role` because this role will by default be authenticated by api-server, otherwise we get into whole spaghetti of `aws-auth` config map for authentication.

In the new solution, we will create another IAM role inside `soak/bootstrap.sh`, which will be used for running `eksctl create cluster` command and then we will store this IAM role in the SSM(to gain access to soak cluster) and the IRSA role(for installing ACK controller on soak cluster).

This week, I will write more documentation inside test-infra to describe this behavior and make above mentioned changes as well. But please remove this Step3 for now.

Reply:
Removing step 3 for now. I wonder if we could find some way to reduce the complexity here - by creating IAM roles with a standard name and using a service principal trust relationships - so we don't need to store stuff in SSM or mess with annoying kubeconfig messes.